packages/react-i18n/docs/react_i18n_schema.md

```diff
@@ -105,48 +120,116 @@ In order to avoid key collisions, the following keys are reserved for use as plu
+### Invalid cardinal pluralization keys
 
-Furthermore, pluralization keys that are not required by the language must not be used, even within a pluralization context.
+Furthermore, cardinal pluralization keys that are not required by the language must not be used, even within a cardinal pluralization context.
 
-For example, even though `zero` is a required pluralization key for some languages (eg. Arabic), it is not a required pluralization key in English. Therefore, the `zero` key must not be present in an English file alongside the pluralization KVPs required by English.
+For example, even though `zero` is a required cardinal pluralization key for some languages (eg. Arabic), it is not a required cardinal pluralization key in English. Therefore, the `zero` key must not be present in an English file alongside the cardinal pluralization KVPs required by English.
 
 **Example INVALID English file:**
 ```jsonc
 {
   "cars": {
-    "zero": "I have no cars", // 🚫 INVALID: `zero` is not a required pluralization key for English!
+    "zero": "I have no cars", // 🚫 INVALID: `zero` is not a required cardinal pluralization key for English!
     "one": "I have {count} car",
     "other": "I have {count} cars"
   }
 }
 ```
 
-## Detecting pluralization contexts
+### Detecting cardinal pluralization contexts
+
+These constraints allow for a simple algorithm for identifying cardinal pluralization contexts:
+
+* If a node has child leaf KVPs for **any** of the cardinal pluralization keys required by the language, outside of a then its **immediate** children are within a cardinal pluralization context.
+
+**Note**: in order to be a valid cardinal pluralization context, any other required cardinal pluralization keys must also be present within the cardinal pluralization context.
+
+## Ordinal Pluralization
+
+Pluralization of ordinal numbers is handled by providing the ordinal pluralization keys required by the language as leaf KVPs, under an `ordinal` parent, creating an "ordinal pluralization context"
+
+**Example English file with ordinal pluralization keys:**
+
+```jsonc
+{
+  "cars": {
+    "ordinal": { // The special key "ordinal" creates an ordinal pluralization context
+      "one": "This is my {amount}st car", // Used for 1, 21, 31, 41, 51, 61, 71, 81, 101, 1001, …
+      "two": "This is my {amount}nd car", // Used for 2, 22, 32, 42, 52, 62, 72, 82, 102, 1002, …
+      "few": "This is my {amount}rd car", // Used for 3, 23, 33, 43, 53, 63, 73, 83, 103, 1003, …
+      "other": "This is my {amount}th car" // Used for 0, 4~18, 100, 1000, 10000, 100000, 1000000, …
+    }
+  }
+}
+```
+
+Different languages require different sets of ordinal pluralization keys. For example, Polish (`pl`) only has 1 required ordinal pluralization key: `other`. The Polish version of the previous example would be:
+
+**Example Polish file with ordinal pluralization keys:**
+
+```json
+{
+  "cars": {
+    "ordinal": {
+      "other": "To mój {amount}. samochód"
```

Comment:
Is the period after the number intentional?

Reply:
Yes, that is how Polish does it: https://en.wikipedia.org/wiki/Ordinal_indicator#Ordinal_dot
I also confirmed it with a native Polish speaker.